Tools/Gulp/tasks/gulpTasks-testsES6.js

```diff
@@ -19,31 +19,24 @@ var es6TestsWebpackFile = path.resolve(es6TestsFolder, "webpack.config.js");
  * Launches the ES6 modules tests to evaluate the min package size.
  */
 gulp.task("tests-es6Modules", function(done) {
-    colorConsole.log("Npm link dependencies");
-    shelljs.exec("npm link @babylonjs/core", {
-        async: false,
-        cwd: es6TestsFolder
-    });
-
-    shelljs.exec("npm link @babylonjs/materials", {
-        async: false,
-        cwd: es6TestsFolder
-    });
-
-    colorConsole.log("Bundle test app");
-    var result = shelljs.exec("npx webpack", {
-        async: false,
-        cwd: es6TestsFolder
-    });
-
-    if (result.code != 0) {
-        colorConsole.error(result.stdout);
-        colorConsole.error(result.stderr);
-        throw "Can not build es6 dev apps."
+    const savedShellConfig = {...shelljs.config};
+    try {
+        Object.assign(shelljs.config, {verbose: true, silent: false, fatal: true});
+
+        colorConsole.log("Link to dev copy of built modules");
+        const coreDevDir = config.core.computed.packageES6DevDirectory;
+        const materialsDevDir = config.materialsLibrary.computed.packageES6DevDirectory;
+        shelljs.exec(`npm install --no-save "${coreDevDir}"`, {cwd: es6TestsFolder});

```

Comment:
would prefer to keep the npm link here and bypassing the task as you do later

Reply:
Hmm, not sure quite how that would work.

Please pardon me if I'm over-explaining:

Previously, `prepareAdditionalDevPackages`, `prepareEs6DevPackages` and `prepareUMDDevPackages` (the dependency packages) would execute `npm link` calls with no arguments. Those creates symlinks from the global environment to the package directories (`packageDevPath` for each package). Then, `tests-es6Modules` (the user of those packages) would execute `npm link @babylonjs/core` and `npm link @babylonjs/materials`, which would create a symlink from the local `node_modules` to the global environment. So you end up with a chain from local `node_modules` => global environment => `packageDevPath` directory.

With this change, to avoid the global environment in the middle, `tests-es6Modules` (the user) directly links to the `packageDevPath` directory for the packages it depends on. This is done using `npm install` (which when given a directory, makes a symlink). That way it doesn't matter whether or not a global environment link was ever made for the other packages -- the dependency will always be to (what should be?) the right place. That _seems_ generally superior to me but I could be missing some use cases!

If it were up to me, I'd just unconditionally remove the `npm link` from the other packages, but I figured people found some benefit in having them globally linked and I didn't want to rock the boat too much.

So... I _could_ check `noGlobalInstall` here in `tests-es6Modules` (the user), and if set use `npm install ${coreDevDir}`, and if not set use `npm link @babylonjs/core` (and similarly for materials). But given that the `npm install` technique should always work and seems more robust, it seemed simpler and better to just always use `npm install` here at the user. (Also it avoids needing to share the flag value.)

But I'm sure I'm missing something?

Reply:
(Updated to check noInstallGlobal, and choose "npm link" if not set.)